### DISCUSSION TITLE
Pure

### DISCUSSION THREAD

**Discussion by Sonderwoods** [mariuszhermansdorfer/SandWorm#95] [Q&A]
https://github.com/mariuszhermansdorfer/SandWorm/blob/6721afb1168c5276cdd99c7fe0878e0256503c96/SandWorm/GaussianBlurProcessor.cs#L51

Hey @mariuszhermansdorfer ,  i saw the pure attribute and immediately looked it up. Are you using the attribute for any code analyzer, or just to tell yourself that this method is actually pure? Or other reasons to use the attribute?

**Comment by mariuszhermansdorfer** [comment]
Back then we were toying with the idea of using [code contracts](https://docs.microsoft.com/en-us/dotnet/framework/debug-trace-profile/code-contracts) to optimize this logic even further. It turned out to work very fast as is without further code changes and we decided to leave it as is. This attribute doesn't do much in the current context.
